Data collection errors related to secondary database availability.

### Component

SQL collection scripts

### Performance Monitor Version

2.0.0

### SQL Server Version

Microsoft SQL Server 2016 (SP3-GDR) (KB5068401) - 13.0.6475.1 (X64)   Oct  4 2025 05:49:46   Copyright (c) Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor) 

### Windows Version

Windows Server 2016 Standard

### Describe the Bug

Getting data collection errors due to my secondary database not allowing read only intent.

This is how I updated database_configuration_collector to eliminate the errors.  

[alter database_configuration_collector - Copy.txt](https://github.com/user-attachments/files/25585753/alter.database_configuration_collector.-.Copy.txt)

Thank you for the great product.


### Steps to Reproduce

Install product via GUI.  Errors when data collection runs at the end of the process.

### Expected Behavior

I would expect that data collection would ignore secondary databases the don't allow read only intent.

### Actual Behavior

Data collection reports errors attempting to access database that are not readable.

### Error Messages / Log Output

```text
From the installation log:  
[ERROR]   database_configuration_collector: Database: DaDatabase - The target database, 'DaDatabase', is participating in an availability group and is currently not accessible for queries. Either data movement is suspended or the availability replica is not enabled for read access...
```

### Screenshots

_No response_

### Additional Context

_No response_

--- a/Lite/Services/RemoteCollectorService.QueryStore.cs
+++ b/Lite/Services/RemoteCollectorService.QueryStore.cs
@@ -48,6 +48,23 @@ DECLARE db_check CURSOR LOCAL FAST_FORWARD FOR
     AND   d.database_id < 32761
     AND   d.state_desc = N'ONLINE'
     AND   d.name <> N'PerformanceMonitor'
+    AND   d.database_id NOT IN
+          (
+              SELECT
+                  d2.database_id
+              FROM sys.databases AS d2
+              JOIN sys.availability_replicas AS r
+                ON d2.replica_id = r.replica_id
+              WHERE NOT EXISTS
+                    (
+                        SELECT
+                            1/0
+                        FROM sys.dm_hadr_availability_group_states AS s
+                        WHERE s.primary_replica = r.replica_server_name
+                    )
+              AND   r.secondary_role_allow_connections_desc = N'READ_ONLY'
+              AND   r.replica_server_name = @@SERVERNAME
+          )
     OPTION(RECOMPILE);
 
 OPEN db_check;

--- a/Lite/Services/RemoteCollectorService.ServerConfig.cs
+++ b/Lite/Services/RemoteCollectorService.ServerConfig.cs
@@ -336,6 +336,23 @@ WHERE (d.database_id > 4 OR d.database_id = 2)
 AND   d.database_id < 32761
 AND   d.name <> N'PerformanceMonitor'
 AND   d.state_desc = N'ONLINE'
+AND   d.database_id NOT IN
+      (
+          SELECT
+              d2.database_id
+          FROM sys.databases AS d2
+          JOIN sys.availability_replicas AS r
+            ON d2.replica_id = r.replica_id
+          WHERE NOT EXISTS
+                (
+                    SELECT
+                        1/0
+                    FROM sys.dm_hadr_availability_group_states AS s
+                    WHERE s.primary_replica = r.replica_server_name
+                )
+          AND   r.secondary_role_allow_connections_desc = N'READ_ONLY'
+          AND   r.replica_server_name = @@SERVERNAME
+      )
 ORDER BY d.name
 OPTION(RECOMPILE);";
 

--- a/install/09_collect_query_store.sql
+++ b/install/09_collect_query_store.sql
@@ -291,6 +291,23 @@ BEGIN
             AND   d.is_read_only = 0
             AND   d.name <> N'PerformanceMonitor'
             AND   d.database_id < 32761 /*exclude contained AG system databases*/
+            AND   d.database_id NOT IN
+                  (
+                      SELECT
+                          d2.database_id
+                      FROM sys.databases AS d2
+                      JOIN sys.availability_replicas AS r
+                        ON d2.replica_id = r.replica_id
+                      WHERE NOT EXISTS
+                            (
+                                SELECT
+                                    1/0
+                                FROM sys.dm_hadr_availability_group_states AS s
+                                WHERE s.primary_replica = r.replica_server_name
+                            )
+                      AND   r.secondary_role_allow_connections_desc = N'READ_ONLY'
+                      AND   r.replica_server_name = @@SERVERNAME
+                  )
             OPTION(RECOMPILE);
 
         OPEN @db_check_cursor;

--- a/install/39_collect_database_configuration.sql
+++ b/install/39_collect_database_configuration.sql
@@ -165,6 +165,23 @@ BEGIN
             AND   d.name != DB_NAME()
             AND   d.state_desc = N'ONLINE'
             AND   d.database_id < 32761 /*exclude contained AG system databases*/
+            AND   d.database_id NOT IN
+                  (
+                      SELECT
+                          d2.database_id
+                      FROM sys.databases AS d2
+                      JOIN sys.availability_replicas AS r
+                        ON d2.replica_id = r.replica_id
+                      WHERE NOT EXISTS
+                            (
+                                SELECT
+                                    1/0
+                                FROM sys.dm_hadr_availability_group_states AS s
+                                WHERE s.primary_replica = r.replica_server_name
+                            )
+                      AND   r.secondary_role_allow_connections_desc = N'READ_ONLY'
+                      AND   r.replica_server_name = @@SERVERNAME
+                  )
             ORDER BY
                 d.name
             OPTION (RECOMPILE);

